music/src/music_rnn/model.ts

```diff
@@ -51,14 +70,49 @@ export class MusicRNN {
 
   private initialized: boolean;
 
+  /**
+   * Create a `MusicRNN` object from URL. If `spec` not provided, loads a
+   * `config.json` file containing model specification from the URL directory.
+   * Does not initialize the model weights.
+   *
+   * @param url Path to a checkpoint directory.
+   * @param spec (Optional) `MusicRNNSpec` object. If undefined, will be loaded
+   * from a `config.json` file in the checkpoint directory.
+   */
+  static async fromURL(url: string, spec?: MusicRNNSpec) {
+    function modelFromSpec(spec: MusicRNNSpec) {
+      const dataConverter = data.converterFromSpec(spec.dataConverter);
+      const chordEncoder = spec.chordEncoder ?
+          chords.chordEncoderFromType(spec.chordEncoder) :
+          undefined;
+      const auxInputs = spec.auxInputs ?
+          spec.auxInputs.map(s => aux_inputs.auxiliaryInputFromSpec(s)) :
+          undefined;
+      return new MusicRNN(
+          url, dataConverter, spec.attentionLength, chordEncoder, auxInputs);
+    }
+    if (spec) {
+      return modelFromSpec(spec);
+    } else {
+      return await fetch(`${url}/config.json`)
+          .then((response) => response.json())
+          .then((spec) => {
+            if (spec.type !== 'MusicRNN') {
+              throw new Error(
```

Comment:
Can you put this in modelFromSpec?

Reply:
Or, are you wanting to allow type to not be specified if passed via the constructor? I could go either way.

Reply:
If it's passed via constructor, doesn't the compile-time check suffice?